### ISSUE TITLE
No signature of method: org.apache.tools.zip.ZipOutputStream.setUseZip64()

### ISSUE THREAD

**Issue by baohuahuang1** [johnrengelman/shadow#188]
I am on gradle 2.3, and when I use the shadow plugin I get the following exception:

Caused by: groovy.lang.MissingMethodException: No signature of method: org.apache.tools.zip.ZipOutputStream.setUseZip64() is applicable for argument types: (org.apache.tools.zip.Zip64Mode) values: [Never]
    at com.github.jengelman.gradle.plugins.shadow.internal.DefaultZipCompressor.createArchiveOutputStream(DefaultZipCompressor.groovy:35)
    ... 28 more

Any idea how to fix this?
Thanks
Baohua

**Comment by johnrengelman** [comment]
Can you update to a more recent version of Gradle?
Also which version of shadow are you running?

**Comment by datallah** [comment]
We've seen this happen when another dependency pulled in an old version of ant (using the older 'ant:ant:$VERSION' maven group/artifact) in addition to the 'org.apache.ant:ant:1.9.3' pulled in by the shadowjar plugin.

**Comment by johnrengelman** [comment]
Interesting...thanks @datallah 
Probably another point for packing `shadow` with itself...

**Comment by rahulsom** [comment]
Here's my dependencies for non test and prod scopes

```

archives - Configuration for archive artifacts.
No dependencies

asciidoctor
+--- org.asciidoctor:asciidoctorj:1.5.3.2
|    +--- org.jruby:jruby-complete:1.7.21
|    \--- com.beust:jcommander:1.35
\--- org.asciidoctor:asciidoctorj-groovy-dsl:1.0.0.preview2
     +--- org.codehaus.groovy:groovy-all:1.8.9
     \--- org.asciidoctor:asciidoctorj:1.5.1 -> 1.5.3.2 (*)

cobertura
\--- net.sourceforge.cobertura:cobertura:2.1.1
     +--- org.ow2.asm:asm:5.0.1
     +--- org.ow2.asm:asm-tree:5.0.1
     |    \--- org.ow2.asm:asm:5.0.1
     +--- org.ow2.asm:asm-commons:5.0.1
     |    \--- org.ow2.asm:asm-tree:5.0.1 (*)
     +--- org.ow2.asm:asm-util:5.0.1
     |    \--- org.ow2.asm:asm-tree:5.0.1 (*)
     +--- org.ow2.asm:asm-analysis:5.0.1
     |    \--- org.ow2.asm:asm-tree:5.0.1 (*)
     +--- oro:oro:2.0.8
     +--- jaxen:jaxen:1.1.4
     +--- org.apache.ant:ant:1.8.3
     |    \--- org.apache.ant:ant-launcher:1.8.3
     +--- org.apache.ant:ant-junit:1.8.3
     |    +--- org.apache.ant:ant:1.8.3 (*)
     |    \--- junit:junit:3.8.2 -> 4.11
     |         \--- org.hamcrest:hamcrest-core:1.3
     +--- org.mortbay.jetty:servlet-api-2.5:6.1.14
     +--- org.mortbay.jetty:jetty:6.1.14
     |    +--- org.mortbay.jetty:jetty-util:6.1.14
     |    \--- org.mortbay.jetty:servlet-api-2.5:6.1.14
     +--- org.mortbay.jetty:jetty-util:6.1.14
     +--- ch.qos.logback:logback-classic:1.0.13
     |    \--- ch.qos.logback:logback-core:1.0.13
     +--- junit:junit:4.11 (*)
     \--- org.apache.commons:commons-lang3:3.3.2

codenarc - The CodeNarc libraries to be used for this project.
+--- org.gmetrics:GMetrics:0.7
|    +--- org.codehaus.groovy:groovy:[2.1.0,) -> 2.4.6
|    +--- org.codehaus.groovy:groovy-xml:[2.1.0,) -> 2.4.6
|    |    \--- org.codehaus.groovy:groovy:2.4.6
|    \--- org.codehaus.groovy:groovy-ant:[2.1.0,) -> 2.4.6
|         +--- org.codehaus.groovy:groovy:2.4.6
|         +--- org.codehaus.groovy:groovy-groovydoc:2.4.6
|         |    +--- org.codehaus.groovy:groovy-templates:2.4.6
|         |    |    +--- org.codehaus.groovy:groovy-xml:2.4.6 (*)
|         |    |    \--- org.codehaus.groovy:groovy:2.4.6
|         |    \--- org.codehaus.groovy:groovy:2.4.6
|         +--- org.apache.ant:ant-antlr:1.9.4
|         \--- org.apache.ant:ant-junit:1.9.4
|              \--- junit:junit:4.11
|                   \--- org.hamcrest:hamcrest-core:1.3
+--- net.sourceforge.cobertura:cobertura:2.1.1
|    +--- org.ow2.asm:asm:5.0.1
|    +--- org.ow2.asm:asm-tree:5.0.1
|    |    \--- org.ow2.asm:asm:5.0.1
|    +--- org.ow2.asm:asm-commons:5.0.1
|    |    \--- org.ow2.asm:asm-tree:5.0.1 (*)
|    +--- org.ow2.asm:asm-util:5.0.1
|    |    \--- org.ow2.asm:asm-tree:5.0.1 (*)
|    +--- org.ow2.asm:asm-analysis:5.0.1
|    |    \--- org.ow2.asm:asm-tree:5.0.1 (*)
|    +--- oro:oro:2.0.8
|    +--- jaxen:jaxen:1.1.4
|    +--- org.apache.ant:ant-junit:1.8.3 -> 1.9.4 (*)
|    +--- org.mortbay.jetty:servlet-api-2.5:6.1.14
|    +--- org.mortbay.jetty:jetty:6.1.14
|    |    +--- org.mortbay.jetty:jetty-util:6.1.14
|    |    \--- org.mortbay.jetty:servlet-api-2.5:6.1.14
|    +--- org.mortbay.jetty:jetty-util:6.1.14
|    +--- ch.qos.logback:logback-classic:1.0.13
|    |    \--- ch.qos.logback:logback-core:1.0.13
|    +--- junit:junit:4.11 (*)
|    \--- org.apache.commons:commons-lang3:3.3.2
\--- org.codenarc:CodeNarc:0.25.2
     +--- junit:junit:4.8.1 -> 4.11 (*)
     +--- org.codehaus.groovy:groovy-xml:2.1.8 -> 2.4.6 (*)
     +--- org.codehaus.groovy:groovy:2.1.8 -> 2.4.6
     +--- org.codehaus.groovy:groovy-ant:2.1.8 -> 2.4.6 (*)
     \--- org.gmetrics:GMetrics:0.7 (*)

shadow
No dependencies
```

I get the same exception

```
:shadowJar FAILED
:lintGradle

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':shadowJar'.
> Could not create ZIP '/Users/rahul/src/fry/build/libs/fry-1.9999-SNAPSHOT.jar'

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':shadowJar'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:68)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
	at org.gradle.execution.taskgraph.ParallelTaskPlanExecutor.process(ParallelTaskPlanExecutor.java:47)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
	at org.gradle.internal.Factories$1.create(Factories.java:22)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:81)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:46)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:52)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.DaemonHealthTracker.execute(DaemonHealthTracker.java:47)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.HintGCAfterBuild.execute(HintGCAfterBuild.java:41)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:246)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: org.gradle.api.GradleException: Could not create ZIP '/Users/rahul/src/fry/build/libs/fry-1.9999-SNAPSHOT.jar'
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction.execute(ShadowCopyAction.groovy:71)
	at org.gradle.api.internal.file.copy.NormalizingCopyActionDecorator.execute(NormalizingCopyActionDecorator.java:52)
	at org.gradle.api.internal.file.copy.DuplicateHandlingCopyActionDecorator.execute(DuplicateHandlingCopyActionDecorator.java:42)
	at org.gradle.api.internal.file.copy.CopyActionExecuter.execute(CopyActionExecuter.java:38)
	at org.gradle.api.tasks.AbstractCopyTask.copy(AbstractCopyTask.java:83)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar.copy(ShadowJar.java:70)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:228)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:221)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:210)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 68 more
Caused by: org.gradle.api.UncheckedIOException: Unable to create ZIP output stream for file /Users/rahul/src/fry/build/libs/fry-1.9999-SNAPSHOT.jar.
	at com.github.jengelman.gradle.plugins.shadow.internal.DefaultZipCompressor.createArchiveOutputStream(DefaultZipCompressor.groovy:41)
	at com.github.jengelman.gradle.plugins.shadow.internal.ZipCompressor$createArchiveOutputStream.call(Unknown Source)
	at com.github.jengelman.gradle.plugins.shadow.tasks.ShadowCopyAction.execute(ShadowCopyAction.groovy:69)
	... 81 more
Caused by: groovy.lang.MissingMethodException: No signature of method: org.apache.tools.zip.ZipOutputStream.setUseZip64() is applicable for argument types: (org.apache.tools.zip.Zip64Mode) values: [Never]
	at com.github.jengelman.gradle.plugins.shadow.internal.DefaultZipCompressor.createArchiveOutputStream(DefaultZipCompressor.groovy:36)
	... 83 more


BUILD FAILED

Total time: 12.154 secs

```

I'm using gradle 2.13 with shadow plugin 1.2.3

**Comment by johnrengelman** [comment]
@rahulsom those dependencies won't show you the problem. You need to get the dependencies that are on the buildscript classpath.

**Comment by rahulsom** [comment]
How do I get that report?

**Comment by rahulsom** [comment]
In my `buildSrc`, I had this

```groovy
dependencies {
  compile 'org.cassandraunit:cassandra-unit:3.0.0.1'
  compile 'com.datastax.cassandra:cassandra-driver-core:3.0.1'
}
```

I turned it into 

```groovy
dependencies {
  compile 'org.apache.ant:ant:1.9.6'
  compile 'org.cassandraunit:cassandra-unit:3.0.0.1'
  compile 'com.datastax.cassandra:cassandra-driver-core:3.0.1'
}
```

Also in my testCompile, I had this

```groovy
  testCompile 'org.cassandraunit:cassandra-unit:3.0.0.1'
```

I turned it into

```groovy
  testCompile 'org.apache.ant:ant:1.9.6'
  testCompile 'org.cassandraunit:cassandra-unit:3.0.0.1'
```

Now it works. 

However this was too much blind coding. If you know how to detect the dependency that was polluting the shadowJar classpath, I would be glad to learn.

Thanks for your help @johnrengelman !

**Comment by johnrengelman** [comment]
I'd search the Gradle forums on how to get the build script class path.

**Comment by Almighty-Alpaca** [comment]
I just experienced the same problem. I got it while using the plugin in a subproject (and only there). I noticed that on my classpath of the root project there is ant tools verion 1.8.0 instead of 1.9.4 which didn't had the method.
My root project's classpath:
```
com.jfrog.bintray.gradle:gradle-bintray-plugin:1.7.1
org.codehaus.groovy.modules.http-builder:http-builder:0.7.2
org.apache.httpcomponents:httpclient:4.2.1
org.apache.httpcomponents:httpcore:4.2.1
commons-logging:commons-logging:1.1.1
commons-codec:commons-codec:1.6
net.sf.json-lib:json-lib:2.3
commons-beanutils:commons-beanutils:1.8.0
commons-collections:commons-collections:3.2.1
commons-lang:commons-lang:2.4
net.sf.ezmorph:ezmorph:1.0.6
net.sourceforge.nekohtml:nekohtml:1.9.16
xerces:xercesImpl:2.9.1
xml-resolver:xml-resolver:1.2
org.apache.maven:maven-ant-tasks:2.1.3
org.apache.ant:ant:1.8.0
org.apache.ant:ant-launcher:1.8.0
classworlds:classworlds:1.1-alpha-2
org.codehaus.plexus:plexus-container-default:1.0-alpha-9-stable-1
junit:junit:3.8.1
org.codehaus.plexus:plexus-utils:1.5.15
org.codehaus.plexus:plexus-interpolation:1.11
org.apache.maven:maven-artifact:2.2.1
org.apache.maven:maven-artifact-manager:2.2.1
org.apache.maven:maven-repository-metadata:2.2.1
org.apache.maven.wagon:wagon-provider-api:1.0-beta-6
backport-util-concurrent:backport-util-concurrent:3.1
org.apache.maven:maven-model:2.2.1
org.apache.maven:maven-project:2.2.1
org.apache.maven:maven-settings:2.2.1
org.apache.maven:maven-profile:2.2.1
org.apache.maven:maven-plugin-registry:2.2.1
org.apache.maven:maven-error-diagnostics:2.2.1
org.apache.maven.wagon:wagon-file:1.0-beta-6
org.apache.maven.wagon:wagon-http-lightweight:1.0-beta-6
org.apache.maven.wagon:wagon-http-shared:1.0-beta-6
nekohtml:xercesMinimal:1.9.6.2
nekohtml:nekohtml:1.9.6.2
com.github.ben-manes:gradle-versions-plugin:0.13.0
com.thoughtworks.xstream:xstream:1.4.7
xmlpull:xmlpull:1.1.3.1
xpp3:xpp3_min:1.1.4c
```
On the subproject's classpath there is the correct version though:
```
com.github.jengelman.gradle.plugins:shadow:1.2.3
org.jdom:jdom2:2.0.5
org.ow2.asm:asm:5.0.3
org.ow2.asm:asm-commons:5.0.3
org.ow2.asm:asm-tree:5.0.3
commons-io:commons-io:2.4
org.apache.ant:ant:1.9.4
org.apache.ant:ant-launcher:1.9.4
org.codehaus.plexus:plexus-utils:2.0.6
org.codehaus.groovy:groovy-backports-compat23:2.4.4
```

**Comment by Almighty-Alpaca** [comment]
I just fixed it for me by including an updated version of apache ant in my root project's classpath:
```gradle
buildscript {
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath 'org.apache.ant:ant:1.9.7'
	}
}
```

**Comment by johnrengelman** [comment]
This is addressed in v2.0.0 since Shadow packages its own dependencies. That should avoid this conflict.
